tests/kernel/threads/dynamic_thread_stack/src/main.c

```diff
@@ -114,10 +114,27 @@ ZTEST(dynamic_thread_stack, test_dynamic_thread_stack_alloc)
 	/* spwan our threads */
 	for (size_t i = 0; i < N; ++i) {
 		tflag[i] = false;
+#ifdef CONFIG_XTENSA
+		/**
+		 * The stack has to be in the cached region because of
+		 * performance reasons. The heap is not allocated in incoherent
+		 * region but the architecture maps addressable RAM twice in
+		 * two different regions in a way that for any given pointer,
+		 * it is possible to convert it to/from a cached version.
+		 *
```

Comment:
This is closer but still not quite right: the trick only works if the stack size and alignment is padded to fit within an integer number of cache lines, with no overlap on either side that might be used (e.g. for heap metadata) from another core.  Or alternatively (since the allocation is done in subsystem code) you can trim out the cache-aligned subrange of the stack memory and use that.

Reply:
That makes sense. Thinking about it a little bit more, isn't it a general requirement for targets with incoherent memory ? I mean even for static stacks, they should be cache aligned right, otherwise when flushing them we may be overwriting adjacent data in other cores. I don't remember have seen this, will take a look. Or I am just wrong about it :)